### PR TITLE
Implement serialization for StarkProof

### PR DESCRIPTION
**Pull request by matias-gonz**

### CODE REVIEW

**Review by diegokingston — APPROVED** [review]

File: src/starks/utils.rs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    for _ in 0..merkle_path_len {
+        let commitment = bytes[..32]
+            .try_into()
+            .map_err(|_| DeserializationError::InvalidAmountOfBytes)?;
+        merkle_path.push(commitment);
+        bytes = &bytes[32..];
+    }
+
+    Ok((Proof { merkle_path }, bytes))
+}
```
Comment (MauroToscano) [inline src/starks/utils.rs:34]: Can this be implemented as a Serializable method and tested with different number of queries ?

**Comment by MauroToscano** [comment]
We need to add tests with different queries, and make them non stochastic, the rest is good

**Review by MauroToscano — CHANGES_REQUESTED** [review]
We need to add tests with different queries, and make them non stochastic, the rest is good

**Review by pablodeymo — APPROVED** [review]
LGTM

**Review by Oppen — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -16,6 +16,7 @@ bincode = { version = "2.0.0-rc.2", tag = "v2.0.0-rc.2", git = "https://github.c
 cairo-vm = { git = "https://github.com/lambdaclass/cairo-rs/" }
 sha3 = "0.10.6"
 num-integer = "0.1.45"
+proptest = "1.2.0"
 
 [dev-dependencies]
 hex = "0.4.3"
```
Comment (MauroToscano) [inline Cargo.toml:22]:
```suggestion

[dev-dependencies]
proptest = "1.2.0"
hex = "0.4.3"
```

**Review by MauroToscano — APPROVED** [review]

**Pull request merged**